### ISSUE TITLE
Could not load library cudnn_ops_infer64_8.dll. Error code 126

### ISSUE THREAD

**Issue by Bstrum36** [tensorflow/tensorflow#54784]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>

**System information**
- OS Platform : Windows 10 Pro

- TensorFlow installed from (source or binary): Installed via pip
- TensorFlow version: 2.6.2
- Python version:: 3.6.3
- Installed using pip
- CUDA/cuDNN version: Cuda 11.6
- GPU 1080

Below are the  Environment Variables
![image](https://user-images.githubusercontent.com/37058769/156252235-8d51862f-0fd7-4ae2-a169-3df8302d33df.png)

Variable Name   Variable Value
CUDA_PATH   C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6
CUDA_PATH_V10_1    C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v10.1
CUDA_PATH_V11_6    C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6
CuDnn   C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\include
cudnn_ops_infer64_8.dll    C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\bin
CUPTI   C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\extras\CUPTI\lib64

**Describe the problem**

Tensor flow produces the messages below when beginning the training of the model
2022-03-01 15:02:55.368641: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:185] None of the MLIR Optimization Passes are enabled (registered 2)
Epoch 1/15
Could not load library cudnn_ops_infer64_8.dll. Error code 126
Please make sure cudnn_ops_infer64_8.dll is in your library path!

**There is no cudnn_ops_infer64_8.dll** 

**Provide the exact sequence of commands / steps that you executed before running into the problem**

The python code is below.

model = LeNet.build(width=28, height=28, depth=1, classes=2)
model.compile(loss="binary_crossentropy", optimizer="adam",
	metrics=["accuracy"])

# train the network
print("[INFO] training network...")
H = model.fit(trainX, trainY, validation_data=(testX, testY),
	class_weight=classWeight, batch_size=64, epochs=15, verbose=1)

I'm totally clueless as to what is going on the all was working well until I installed  v11.6
Please advise
Thanks

**Comment by Bstrum36** [comment]
Thanks for the quick response.
I am using:
tf version: 2.6.2
Current Python Version- 3.6.3
tensorflow-2.6 and Python 3.6-3.9 are listed as tested environments

I have CUDA v11.6   in the C:\Program Files\NVIDIA GPU Computing
Toolkit\CUDA path
I tried a new install of the latest tensorflow and the messages stated
that the Requirement already satisfied for each step.

Can you give me a step by step on how to create a test environment.

**Comment by Hpdvn** [comment]
Hi,

I faced the same issue today, and fixed it by removing my currents Cuda and Cndnn installation to replace them by a tested build configuration you can find here : https://www.tensorflow.org/install/source_windows#gpu // for windows.

then i just followed the instructions given by tensorflow on this page : https://www.tensorflow.org/install/gpu.

If ur facing issues while trying to install cuda, its probably because the installer do not handle the fact that a previous cuda installation has been done on our computer, so u will have to uninstall it first, with all its components.

Good luck ! 🐵

**Comment by Bstrum36** [comment]
Thanks for the heads up.  I will give it a try.

**Comment by Bstrum36** [comment]
What version of python are you using?

**Comment by Hpdvn** [comment]
3.7

**Comment by tilakrayal** [comment]
@Bstrum36 ,
Please take a look at the installation steps from this [link](https://www.tensorflow.org/install) with the compatible tested build [configurations](https://www.tensorflow.org/install/source_windows#gpu) and let us know if you are facing same issue.Thanks

**Comment by patlevin** [comment]
The issue can be resolved by doing this step: [install zlib for windows](https://docs.nvidia.com/deeplearning/cudnn/install-guide/index.html#install-zlib-windows)

Just happened to me a few minutes ago before I stumbled upon the solution. I simply copying `zlibwapi.dll` to the `bin`-directory of the CUDA installation solves the issue.

**Comment by Bstrum36** [comment]
Thanks for the info. I have tried this but here is what I get:

zlib123dllx64.zip
I unzip and get:
[image: image.png]
I copy the zlibwapi.dll   to CUDA/v11.6

There it is.C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\bin
[image: image.png]
Still get  this error:
Could not load library cudnn_ops_infer64_8.dll. Error code 126
Please make sure cudnn_ops_infer64_8.dll is in your library path!

Any ideas

**Comment by patlevin** [comment]
Are you using cuDNN v8.3.2 (January 10th, 2022), for CUDA 11.5?
This should be the version compatible with CUDA v11.6.

**Comment by tilakrayal** [comment]
@Bstrum36 ,
As suggested in the [document](https://www.tensorflow.org/install/source_windows#gpu) please try the re-install the tensorflow stable version 2.7 or 2.8 with the compatible CUDA and Cudnn.It helps.Thanks!
